File: src/main/kotlin/com/huawei/cangjie/psi/CjModifierListOwner.java
package com.huawei.cangjie.psi;

import com.huawei.cangjie.lexer.CjModifierKeywordToken;
import com.intellij.psi.PsiElement;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;


public interface CjModifierListOwner extends PsiElement ,CjElement {
    @Nullable
    CjModifierList getModifierList();

    boolean hasModifier(@NotNull CjModifierKeywordToken modifier);

    void addModifier(@NotNull CjModifierKeywordToken modifier);
    void removeModifier(@NotNull CjModifierKeywordToken modifier);


}

File: src/main/kotlin/com/huawei/cangjie/psi/CjModifierListOwner.kt
package com.huawei.cangjie.psi

import com.huawei.cangjie.descriptors.DescriptorVisibility
import com.huawei.cangjie.lexer.CjModifierKeywordToken
import com.intellij.psi.PsiElement

interface CjModifierListOwner : PsiElement, CjAnnotated {

    val modifierList: CjModifierList?

    fun hasModifier(modifier: CjModifierKeywordToken): Boolean

    fun addModifier(modifier: CjModifierKeywordToken)

    fun removeModifier(modifier: CjModifierKeywordToken)

    val modifierVisibility: DescriptorVisibility?
}
